components/function-controller/pkg/apis/serverless/v1alpha2/function_defaults.go

```diff
@@ -65,6 +65,11 @@ func (fn *Function) Default(config *DefaultingConfig) {
 
 func (spec *FunctionSpec) defaultReplicas(config *DefaultingConfig, fn *Function) {
 	if spec.Replicas != nil {
+		if spec.ScaleConfig == nil {
```

Comment:
Why do we fill `scaleConfig`, if replicas is set?

Reply:
1- Show the user the setting. 
2- Validation will break if both are removed. So it should be there with a sane value when/if we remove spec.replicas to disable external scaling.

Reply:
1. Why? It will make the output much less readable. Use can see the structure by: reading documentation or by `kubectl explain`.
2. `replicas` is not removed. We can think about defaulting it to 1. 
Additionally this code can introduce bugs when we will decide to create HPA against function as it is mutating `scaleConfig`.

If `scaleConfig` is set to nil, it's very easy to decide to create hpa.

Reply:
1. It's not ideal. The idea is to mix both features in the simplest possible way.
2. We don't have a flag to enable/disable external scaling. The simplest way I could find was to use `replicas` for that.
3. If you create an user-defined HPA against the function, it should work fine, since it will only target the `replicas` filed, as it doesn't use the `scalingConfig` field. However, if we change the controller to create internal HPAs with function targets, we will need to change this code as well. I think that's normal and expected..